tools/zulint/custom_rules.py

```diff
@@ -0,0 +1,291 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import print_function
+from __future__ import absolute_import
+
+import re
+import traceback
+
+from zulint.printer import print_err, colors
+
+if False:
+    from typing import Any, Dict, List, Optional, Tuple, Iterable
+
+    Rule = Dict[str, Any]
+    RuleList = List[Dict[str, Any]]
+    LineTup = Tuple[int, str, str, str]
+
+class CustomRules:
```

Comment:
I have just moved enough code around so that it works. I will still need a lot of finishing but I wanted to get you an idea of what I was talking about. Do you think we should proceed in this direction?

I have moved all the functions that were in `custom_check.py` into this CustomRules class and while still keeping all the specific rules in `custom_check.py`.